onnxruntime/core/providers/rocm/tensor/transpose_impl.cu

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/providers/rocm/cu_inc/common.cuh"
+#include "core/providers/rocm/tensor/transpose_impl.h"
+
+namespace onnxruntime {
+namespace rocm {
+
+template <typename T>
+// __global__ void _TransposeKernel(int32_t shape_rank, const TArray<int64_t> input_strides,
+//                                  const T* __restrict__ input_data, const TArray<fast_divmod> output_strides, T* __restrict__ output_data, HIP_LONG N) {
+__global__ void _TransposeKernel(int32_t shape_rank, const int64_t* input_strides,
```

Comment:
Is there an expected benefit for Transpose3DKernel and Transpose4DKernel, like for CUDA?

Reply:
yes, these kernels will benefit to AMD as well. The ADO task has been created to track it and will be done in a separate PR.